forem/forem
# [Email/Password Authentication] Allow Admins to enable email/password login

## What type of PR is this? (check all applicable)


## Description
This PR gives Admins the ability to enable/disable email login on the user-login page. When enabled, the option to login with email and password is displayed. When disabled, email/password login is not present.

## QA Instructions, Screenshots, Recordings

1. Using 2 incognito browser windows, login as an `admin` user on window1.
2. Navigate to `/admin/config` and toggle `"Allow email password login"` option ON and save configuration.

![enable-email-login](https://user-images.githubusercontent.com/32520970/95513986-33d99880-0989-11eb-9143-c8b8b0c16021.png)

3. In window2, navigate to `http://localhost:xxxx/enter`. You should see the option to login using email and password.

![email-login-enabled](https://user-images.githubusercontent.com/32520970/95514003-3b00a680-0989-11eb-9f2f-dd5e5793889c.png)

4. As `admin` user, toggle `"Allow email password login"` option OFF and save configuration.

![disable-email-login](https://user-images.githubusercontent.com/32520970/95514030-448a0e80-0989-11eb-9623-fb48bbe00e1b.png)

5. In window2, navigate to `http://localhost:xxxx/enter`. You should NOT see the option to login using email and password.

![email-login-disabled](https://user-images.githubusercontent.com/32520970/95514054-4fdd3a00-0989-11eb-9922-cc93de6fbd9d.png)


## Added tests?


## Added to documentation?


## [optional] Are there any post deployment tasks we need to perform?
NO

## [optional] What gif best describes this PR or how it makes you feel?

![AGT Contestant Speaking](https://media.giphy.com/media/iEq5AbQGlVYTAWTAtQ/giphy.gif)

Note: closes https://github.com/forem/InternalProjectPlanning/issues/114
